src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -1,55 +1,72 @@
+
     with open(path) as f:
-        simstat_obj = json.load(f, object_hook=__json_to_simstat)
-
-    return simstat_obj
-
-def __json_to_simstat(d: dict) -> SimStat:
-    print(d)
-    if 'type' in d:
-        if d['type'] == 'Scalar':
-            return Scalar(
-                          value=d.get('value'),
-                          unit=d.get('unit'),
-                          description=d.get('description'),
-                          datatype=d.get('datatype')
-                         )
-        elif d['type'] == 'Distribution':
-            return Distribution(
-                                value=d.get('value'),
-                                min=d.get('min'),
-                                max=d.get('max'),
-                                num_bins=d.get('numBins'),
-                                bin_size=d.get('binSize'),
-                                sum=d.get('sum'),
-                                sum_squared=d.get('sumSquared'),
-                                underflow=d.get('underflow'),
-                                overflow=d.get('overflow'),
-                                logs=d.get('logs'),
-                                unit=d.get('unit'),
-                                description=d.get('description'),
-                                datatype=d.get('datatype')
-                               )
-        elif d['type'] == 'Accumulator':
-            return Accumulator(
-                                value=d.get('value'),
-                                count=d.get('count'),
-                                min=d.get('min'),
-                                max=d.get('max'),
-                                sumSquared=d.get('sum_squared'),
-                                unit=d.get('unit'),
-                                description=d.get('description'),
-                                datatype=d.get('datatype')
-                               )
-        elif d['type'] == 'Group':
-            return Group(**d)
-        elif d['type'] == 'Vector':
-            return Group(**d)
-    else:
-        return SimStat(**d)
\ No newline at end of file
+    simstat_object = json.load(f, cls=JSONLoader)
+    ```
+    """
+
+    def __init__(self):
+        json.JSONDecoder.__init__(self, object_hook=self.__json_to_simstat)
+
+    def __json_to_simstat(self, d: dict) -> SimStat:
+        if 'type' in d:
+            if d['type'] == 'Scalar':
+                d.pop('type', None)
+                return Scalar(**d)
+
+            elif d['type'] == 'Distribution':
+                d.pop('type', None)
+                return Distribution(**d)
+
+            elif d['type'] == 'Accumulator':
+                d.pop('type', None)
+                return Accumulator(**d)
+
+            elif d['type'] == 'Group':
+                return Group(**d)
+
+            elif d['type'] == 'Vector':
+                d.pop('type', None)
+                d.pop('time_conversion', None)
```

Comment:
d.pop() is not ideal, but it works

Reply:
@BobbyRBruce, why does vector not have a time conversion, but everything else does...?

Reply:
It does have time_conversation. Vector inherets from "Group" which contains it.

Reply:
So, why `pop('time_conversion')`, @yllekn?

Reply:
The constructor for Vector has 'time_conversion = None' in it. I'm not sure what the intent was, or if it was meant to be an optional argument, but it's the same issue as before with 'type'-- setting the property 'time_conversion' twice.

Reply:
In other words, I get that same 'multiple values' error.